src/index.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import escapeRegExp from 'lodash.escaperegexp';
+
+import match from './match/match';
+import convertKeywordsToCriteria from './utils/convertKeywordsToCriteria';
+import ensureObjectOfRegExps from './utils/ensureObjectOfRegExps';
+import parseKeywords from './utils/parseKeywords';
+
+export interface OptionsType {
+  keywords?: string[] | string | null;
+  limit?: number;
+  caseSensitive?: boolean;
+  index?: boolean;
+  insensitive?: string;
+  predicate?: string;
+  ignorePaths?: RegExp[] | string[];
+  pathAlias?: { abc: string | RegExp } | { [s: string]: string | RegExp };
+}
+export interface Criterion {
+  is: boolean | RegExp;
+  key: boolean | RegExp | string;
+  negate: boolean;
+  valueReg: boolean | undefined;
+  checkString: (arg: string) => boolean;
+  checkNumber: (arg: number) => boolean;
+}
+export interface ArrayType {
```

Comment:
What is this type for??

Reply:
https://github.com/cheminfo/smart-array-filter/blob/9e8b69a246556deb7cf769a0ddcd146bf993bbb0/src/__tests__/index.test.js#L5-L30
ArrayType is for this

Reply:
Deleted

Reply:
Deleted